src/java.base/share/classes/java/time/Instant.java

```diff
@@ -1300,12 +1300,9 @@ public boolean equals(Object otherInstant) {
         if (this == otherInstant) {
             return true;
         }
-        if (otherInstant instanceof Instant) {
-            Instant other = (Instant) otherInstant;
-            return this.seconds == other.seconds &&
-                   this.nanos == other.nanos;
-        }
-        return false;
+        return (otherInstant instanceof Instant other)
+                && this.seconds == other.seconds
+                && this.nanos == other.nanos;
     }
```

Comment:
Ditto, `otherInstance` is not strictly always an instant and it would be more consistent to swap the names.
`(other instanceof Instant otherInstant)`.

Reply:
Parameter and pattern variable names swapped, as suggested. See 647bd6b